models/classes/StatisticalMetadata/Import/Validator/HeaderValidator.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA.
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\model\StatisticalMetadata\Import\Validator;
+
+use RuntimeException;
+use InvalidArgumentException;
+use oat\tao\model\StatisticalMetadata\Contract\Header;
+use oat\tao\model\StatisticalMetadata\Import\Mapper\StatisticalMetadataMapper;
+use oat\tao\model\StatisticalMetadata\Import\Extractor\MetadataHeadersExtractor;
+
+class HeaderValidator
+{
+    private const ALLOWED_WIDGETS = [
+        'http://www.tao.lu/datatypes/WidgetDefinitions.rdf#TextBox',
+        'http://www.tao.lu/datatypes/WidgetDefinitions.rdf#TextArea',
+        'http://www.tao.lu/datatypes/WidgetDefinitions.rdf#HTMLArea',
+    ];
+
+    public function __construct(MetadataHeadersExtractor $metadataHeadersExtractor)
+    {
+        $this->metadataHeadersExtractor = $metadataHeadersExtractor;
+    }
+
+    /** @var MetadataHeadersExtractor */
+    private $metadataHeadersExtractor;
```

Comment:
Please put the properties before the first method (the constructor) :)

```suggestion
    /** @var MetadataHeadersExtractor */
    private $metadataHeadersExtractor;
    
    public function __construct(MetadataHeadersExtractor $metadataHeadersExtractor)
    {
        $this->metadataHeadersExtractor = $metadataHeadersExtractor;
    }
```

Reply:
Nice, seems like IDE generated property with an issue 😮 
Thanks!